### ISSUE TITLE
Sign-in stuck on latest builds of v2 on MacOS

### ISSUE THREAD

**Issue by gotgenes** [ytmdesktop/ytmdesktop#1221]
### Consent

- [X] I verified that there is no open/closed issue for the same subject.
- [X] I understand that **YTMDesktop have NO affiliation with Google or YouTube**

### Current Behavior

With 2-Step Verification enabled for my account, when I sign in to YTMD with my credentials, then enter my MFA, the app gets stuck at the end of sign in. Although the Next button appears, and can be clicked, it does not proceed.

![Screenshot 2023-12-18 at 10 07 35 AM](https://github.com/ytmdesktop/ytmdesktop/assets/8517/2526e8f9-8fca-4a0f-8ae1-aca755799cdb)

I get this behavior with both the x86_64 and the arm64 builds of  3df63fa which I obtained from the [build actions artifacts](https://github.com/ytmdesktop/ytmdesktop/actions/runs/7148868728).

### Expected Behavior

I expect the sign in process to complete and to be taken to the YouTube Music home screen.

### Steps To Reproduce

1. Enable 2-Step/MFA security on your Google account (not confirmed if MFA is the key part of the problem because I do not wish to disable MFA on my account).
2. Enter your credentials in YTMD.
3. Enter your MFA code or activate your Security Key.
4. Observe the Next button does not let you proceed to the home screen.

### YTMDesktop

2.0-rc5+

### OS

MacOS

### OS Version

14.2

### Arch

x64

### Installation way

.zip

### Anything else?

It would be helpful if there was some documentation that described how to completely wipe all YTMD files, credentials, etc. so I could see if this is reproducible for a system that has no prior state from earlier YTMD versions.

I tried to find as many as I could.

Here are the logs when I signed in after wiping all the files I found.

```
❯ bat -p Logs/YouTube\ Music\ Desktop\ App/main.log
[2023-12-18 10:26:09.208][main][info] Application launched
[2023-12-18 10:26:09.212][main][info] Created memory store
[2023-12-18 10:26:09.227][main][info] Performing store migration from 0.0.0 to >=2.0.0-rc.2
[2023-12-18 10:26:09.246][main][info] Performing store migration from >=2.0.0-rc.2 to >=2.0.0-rc.4
[2023-12-18 10:26:09.259][main][info] Created electron store
[2023-12-18 10:26:09.356][main][info] Application ready
[2023-12-18 10:26:09.356][main][info] Setup IPC handlers
[2023-12-18 10:26:09.367][main][info] Setup permission handlers
[2023-12-18 10:26:09.370][main][info] Unregistered shortcuts
[2023-12-18 10:26:09.370][main][info] Registered shortcuts
[2023-12-18 10:26:09.381][main][info] Created tray icon
[2023-12-18 10:26:09.404][main][info] Created main window
[2023-12-18 10:26:09.406][main][info] Created YTM view
[2023-12-18 10:26:09.409][main][info] Setup taskbar features
[2023-12-18 10:26:09.410][main][info] Integration update: Zoom Factor
[2023-12-18 10:26:09.410][main][info] Starting enabled integrations
[2023-12-18 10:26:10.253][renderer][info] LegacyDataMixin will be applied to all legacy elements.
Set `_legacyUndefinedCheck: true` on element class to enable.
[2023-12-18 10:26:41.317][main][info] Blocking YTM View redirect to https://www.youtube.com/signin_prompt?app=desktop&next=https%3A%2F%2Fmusic.youtube.com%2F
```

Note that I have multiple YouTube accounts. It could be trying to finish the flow where I select which YouTube account I want to use with YouTube Music.
![Screenshot 2023-12-18 at 10 30 42 AM](https://github.com/ytmdesktop/ytmdesktop/assets/8517/cab3f853-1d93-42f4-b80f-7d6fa4842526)

**Comment by Alipoodle** [comment]
So signing out myself and trying to re-login it would behaviour in a similar way (I have a "Brand" and a personal YT account as well)
:) I suspect that we accidentally block this page, and need to add it onto the list of Authorised URLs, thanks for providing some helpful information especially the logs 

I too also got the same problem:
```
[2023-12-18 19:38:52.106][main][info] Blocking YTM View redirect to https://www.youtube.com/signin_prompt?app=desktop&next=https%3A%2F%2Fmusic.youtube.com%2F
[2023-12-18 19:39:12.506][main][info] Blocking YTM View redirect to https://www.youtube.com/signin_prompt?app=desktop&next=https%3A%2F%2Fmusic.youtube.com%2F
```

**Comment by Alipoodle** [comment]
This should be fixed as of eefc04290673c90b81646f19a16ace46eb71175a now ❤️

**Comment by NovusTheory** [comment]
Closing as this is in the v2.0.0 release
